pkg/kubelet/status/status_manager.go

```diff
@@ -650,23 +645,14 @@ func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUp
 	}
 
 	m.podStatuses[pod.UID] = newStatus
+	m.podStatusesNew[pod.UID] = struct{}{}
 
 	select {
-	case m.podStatusChannel <- podStatusSyncRequest{pod.UID, newStatus}:
-		klog.V(5).InfoS("Status Manager: adding pod with new status to podStatusChannel",
-			"pod", klog.KObj(pod),
-			"podUID", pod.UID,
-			"statusVersion", newStatus.version,
-			"status", newStatus.status)
-		return true
+	case m.podStatusChannel <- struct{}{}:
 	default:
-		// Let the periodic syncBatch handle the update if the channel is full.
-		// We can't block, since we hold the mutex lock.
-		klog.V(4).InfoS("Skipping the status update for pod for now because the channel is full",
-			"pod", klog.KObj(pod),
-			"status", status)
-		return false
+		// there's already a status update pending
 	}
+	return true
```

Comment:
Yeah, the bug is that before you could queue a status change and NO one would ever see it and it wouldn't be reconciled.

After my change, TerminatePod is the pod worker handing responsibility off to status manager who guarantees that it will take the responsibility to drive to completion.

Arguably status manager is a function of pod worker vs a separate component but i don't want to get bogged down in that now.

Agree we can remove the bool since now it doesn't mean anything.